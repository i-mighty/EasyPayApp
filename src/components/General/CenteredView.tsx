import styled from 'styled-components/native';
import { Container } from 'native-base';

const CenteredView = styled(Container)`
  justify-content: center;
  align-items: center;
`;

export default CenteredView;
