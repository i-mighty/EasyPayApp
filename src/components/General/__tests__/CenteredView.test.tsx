/*eslint-disable */
import React from 'react';
import { renderWithProps } from '@src/utils/test-utils';
import CenteredView from '../CenteredView';
const props = {
  testID: 'CenteredView',
};

describe('CenteredView', () => {
  it('renders properly ', () => {
    const { baseElement } = renderWithProps(<CenteredView />);
    expect(baseElement).toMatchSnapshot();
  });
  it('centralizes all components', () => {
    const { getByTestId, baseElement } = renderWithProps(
      <CenteredView {...props} />,
    );
    const container = getByTestId(props.testID);
    expect(container).toHaveStyleRule('justify-content', 'center');
    expect(container).toHaveStyleRule('align-items', 'center');
    expect(baseElement).toMatchSnapshot();
  });
});
