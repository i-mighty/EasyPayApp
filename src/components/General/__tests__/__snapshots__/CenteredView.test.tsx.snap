// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenteredView centralizes all components 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1334,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
      testID="CenteredView"
    />
  </View>
</View>
`;

exports[`CenteredView renders properly  1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1334,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    />
  </View>
</View>
`;
